syncLoop.go

```diff
@@ -61,11 +61,48 @@ func (c *Client) Close() error {
 }
 
 func (c *Client) readLoop(ctx context.Context, filter string) {
+	client := c.WithContext(ctx)
 	next := ""
+	nextRetryTime := 0
```

Comment:
Consider just making `nextRetryTime` a `time.Duration` instead of just an `int`.